src/storage/models.rs

```diff
@@ -0,0 +1,261 @@
+use std::iter;
+
+use chrono::{DateTime, Utc};
+use rand::Rng;
+
+const DEFAULT_SLUG_LENGTH: usize = 8;
+
+/// A code snippet
+#[derive(Debug, Default, Eq, PartialEq)]
+pub struct Snippet {
+    /// Slug that uniquely identifies the snippet
+    pub id: String,
+    /// Snippet title. None means that the title was not specified at snippet
+    /// creation time
+    pub title: Option<String>,
+    /// Snippet syntax. None means that the syntax was not specified at snippet
+    /// creation time
+    pub syntax: Option<String>,
+    /// List of revisions. The snippet content is the content of the most recent
+    /// revision. Revisions are sorted by the version number in the
+    /// ascending order.
+    pub changesets: Vec<Changeset>,
+    /// List of tags attached to the snippet
+    pub tags: Vec<String>,
+    /// Timestamp of when the snippet was created
+    pub created_at: Option<DateTime<Utc>>,
+    /// Timestamp of when the snippet was last modified. Defaults to creation
+    /// time
+    pub updated_at: Option<DateTime<Utc>>,
+}
+
+impl Snippet {
+    /// Create a new Snippet.
+    pub fn new(
+        title: Option<String>,
+        syntax: Option<String>,
+        changesets: Vec<Changeset>,
+        tags: Vec<String>,
+    ) -> Self {
+        let mut snippet = Snippet::default();
```

Comment:
**Q:** What `::default()` is? Where is it implemented?

Reply:
It is a method of the `Default` trait (https://doc.rust-lang.org/std/default/trait.Default.html).

The implementation varies from type to type. E.g. for `usize` it will return `0`, for `f32` it will return `0.0`, etc. For structs you can derive the trait `Default` if every field of the struct implements it.

We don't necessarily *need* it, and can either always use the `Snippet::new()` method, or initialize all the fields explicitly. But, I think, it may be handy for creating an "empty" Snippet instance, while `Snippet::new()` produces an instance with all the required fields populated.